edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -360,44 +388,25 @@ private async Task ManageDeployments(V1ServiceList currentServices, V1Deployment
                 });
 
             // Remove the old
-            IEnumerable<Task<V1Status>> removeServiceTasks = servicesRemoved.Select(
-                i =>
-                {
-                    Events.DeletingService(i);
-                    return this.client.DeleteNamespacedServiceAsync(i.Metadata.Name, this.k8sNamespace, new V1DeleteOptions());
-                });
-            await Task.WhenAll(removeServiceTasks);
-
-            IEnumerable<Task<V1Status>> removeDeploymentTasks = deploymentsRemoved.Select(
+            await Task.WhenAll(deploymentsRemoved.Select(
                 d =>
                 {
                     Events.DeletingDeployment(d);
                     return this.client.DeleteNamespacedDeployment1Async(d.Metadata.Name, this.k8sNamespace, new V1DeleteOptions(propagationPolicy: "Foreground"), propagationPolicy: "Foreground");
-                });
-            await Task.WhenAll(removeDeploymentTasks);
+                }));
 
-            // Create the new.
-            IEnumerable<Task<V1Service>> createServiceTasks = newServices.Select(
-                s =>
-                {
-                    Events.CreatingService(s);
-                    return this.client.CreateNamespacedServiceAsync(s, this.k8sNamespace);
-                });
-            await Task.WhenAll(createServiceTasks);
-
-            IEnumerable<Task<V1Deployment>> createDeploymentTasks = newDeployments.Select(
+            // Create the new 
+            await Task.WhenAll(newDeployments.Select(
                 deployment =>
                 {
                     Events.CreatingDeployment(deployment);
                     return this.client.CreateNamespacedDeploymentAsync(deployment, this.k8sNamespace);
-                });
-            await Task.WhenAll(createDeploymentTasks);
+                }));
 
             // Update the existing - should only do this when different.
-            IEnumerable<Task<V1Deployment>> updateDeploymentTasks = deploymentsUpdated.Select(deployment => this.client.ReplaceNamespacedDeploymentAsync(deployment, deployment.Metadata.Name, this.k8sNamespace));
-            await Task.WhenAll(updateDeploymentTasks);
+            await Task.WhenAll(deploymentsUpdated.Select(deployment => this.client.ReplaceNamespacedDeploymentAsync(deployment, deployment.Metadata.Name, this.k8sNamespace)));
```

Comment:
Can we wait on all of these tasks at the end? I don't think we have to wait for all the deletes to finish before the new creations and updates - at least for Deployments.